### PR TITLE
Test also with Python 3 in macOS

### PR DESCRIPTION
**Pull request by tkf**
based on https://github.com/pytest-dev/pytest/blob/799b72cf6f35c4c4c73797303d3f78c12a795f77/.travis.yml#L38-L52

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -19,11 +18,32 @@ env:
...
+    - &test-macos-py2
+      language: julia
+      os: osx
+      env: PYTHON=python
+      julia: 0.6
+    - &test-macos-py3
+      <<: *test-macos-py2
+      before_install:
+        - brew update
+        - brew upgrade python || echo "Ignoring errors..."
```
Comment (tkf) [inline .travis.yml]: Strangely, python3 is correctly installed https://travis-ci.org/JuliaPy/PyCall.jl/jobs/444940334#L1356 even though the command fails https://travis-ci.org/JuliaPy/PyCall.jl/jobs/444940334#L1244

Comment (tkf) [reply .travis.yml]: Actually, we've been using the same trick in PyJulia: https://github.com/JuliaPy/pyjulia/blob/f7a552d2eabbb5045c7182ba056d0867c05830f9/.travis.yml#L49

File: .travis.yml [hunk 2]
```diff
@@ -1,7 +1,6 @@
 language: julia
 os:
     - linux
-    - osx
```
Comment (stevengj) [inline .travis.yml]: why is this line deleted?

Comment (tkf) [reply .travis.yml]: `osx` tests are added via `matrix.include`: https://github.com/JuliaPy/PyCall.jl/pull/602/files#diff-354f30a63fb0907d4ad57269548329e3R20

I think there would be duplicates if we left `osx` here.
